std/machines/small_field/pointer_arith.asm

```diff
@@ -0,0 +1,39 @@
+/// Creates constraints that increments a 32-bits pointer by `amount` bytes.
+///
+/// Preconditions:
+/// - `amount` must be a power of two,
+/// - `amount` must be in range [1, 0x10000],
+/// - `pre_low` must be a multiple of amount,
+/// - `pre_low` must be in range [0, 0xffff].
+/// - `int(pre_high) * 0x10000 + int(pre_low) + amount` must be in range [0, 0xffffffff].
+///
+/// The pointers are given in 2 16-bit limbs. This function returns a set of constraints
+/// ensuring that the following are equal:
+/// - (int(post_high) * 0x10000 + int(post_low))
+/// - (int(pre_high) * 0x10000 + int(pre_low) + amount)
+///
+/// The returned constraints are not automatically added to the constraints set. This is so the
+/// caller will be able to latch-disable them in rows they aren't needed.
+///
+/// This constr function introduces 2 new witness column and some helper constraints.
+let increment_ptr: expr, expr, expr, expr, expr -> Constr[] = constr |
+    amount,
+    pre_high,
+    pre_low,
+    post_high,
+    post_low
+| {
+    // Is the low limb at its maximum value?
+    let carry;
+    carry = std::utils::is_zero(pre_low - (0x10000 - amount));
```

Comment:
Cool!

I would have done
```
force_bool(carry);
pre_low + amount = carry * 2**16 + post_low;
post_high = pre_high + carry;
```

Needs one less column (no need to store the inverse created in `is_zero`), but would assume `post_low` and `post_high` to be range-constrained! Adding two range constraints would typically be more expensive, so I like this solution more.